new blog post: What's New in Lighthouse 6.0 (#2890)
GoogleChrome/web.dev: src/site/content/en/blog/lighthouse-whats-new-6.0/index.md

---
title: "What's New in Lighthouse 6.0"
authors:
  - cjamcl
date: 2020-05-18

> hero: lighthouse-logo.svg

Reviewer: lol responsive svg's:

![image](https://user-images.githubusercontent.com/39191/82106984-32a58780-96d9-11ea-9ee9-c00623ecf563.png)

Reviewer 2: good catch. https://github.com/GoogleChrome/web.dev/issues/2892

Author: was there something to do for this? logo looks kinda big :)

Reviewer 2: I don't think so. The bug i opened was to not do srcsetting on svg but I don't think it's affecting how it's being displayed. Might try a different image if  you want it to be smaller?

services, including [web.dev/measure](https://web.dev/measure/) and [PageSpeed
Insights](https://developers.google.com/speed/pagespeed/insights/).

Lighthouse 6.0 is available immediately on npm and in [Chrome
Canary](https://www.google.com/chrome/canary/). Other Google services that leverage Lighthouse will
update by the end of the month. It will land in Chrome Stable in Chrome 84 (mid-July).

To try the Lighthouse Node CLI, use the following commands:  
```bash
npm install -g lighthouse
lighthouse https://www.example.com --view
```

This version of Lighthouse comes with a large number of changes that are all listed in the
changelog. (**TODO: link to changelog)**. We'll cover the highlights in this article.

- [New metrics](#new-metrics)
- [Performance score update](#score)
- [New audits](#new-audits)
- [Lighthouse CI](#ci)
- [Renamed Chrome DevTools panel](#devtools)
- [Mobile emulation](#emulation)
- [Browser extension](#extension)
- [Budgets](#budgets)
- [Source location links](#source-location)
- [On the horizon](#horizon)
- [Thank you!](#thanks)

## New metrics {: #new-metrics }

> ![image](lighthouse-new-metrics.png)

Reviewer: needs 1px border. might as well be 80%

here's what that'd look like

![image](https://user-images.githubusercontent.com/39191/82106920-e35f5700-96d8-11ea-855c-c2fda88e5ed2.png)

Lighthouse 6.0 introduces three new metrics to the report. Two of these new metrics–Largest
Contentful Paint (LCP) and Cumulative Layout Shift (CLS)–are lab implementations of [Core Web
Vitals](https://web.dev/vitals/).

### Largest Contentful Paint (LCP) {: #lcp }

[Largest Contentful Paint (LCP)](https://www.web.dev/lcp/) is a measurement of perceived loading
experience. It marks the point during page load when the primary–or "largest"–content has loaded and
is visible to the user. LCP is an important complement to First Contentful Paint (FCP) which only
captures the very beginning of the loading experience. LCP provides a signal to developers about how
quickly a user is actually able to see the content of a page. An LCP score below 2.5 seconds is
considered 'Good.'

For more information, watch this deep-dive on LCP by Paul Irish:

> {% YouTube 'diAc65p15ag' %}

Reviewer: I don't know about others but I dont love the YT embed

also it doesnt use youtube-lite, which i added to web.dev in a PR in november. 😭 

personally i'd prefer these sit as text links.

Reviewer 2: > also it doesnt use youtube-lite, which i added to web.dev in a PR in november. 😭

[it was all smooshed together!](https://github.com/GoogleChrome/web.dev/pull/1882/files#diff-87fd67e4e0f3e4c45fd06774bbcdaa8eR294)

I think we'd need someone to do a pass to clean it up and check it in but it's been hard to get it prioritized with web.dev/live and other feature requests coming in 😅 I'll take a note to see if I can get it on the board in case folks have a spare moment in the coming weeks/months.

(TTI). It is a companion metric to TTI and it brings more nuance to quantifying main thread activity
that blocks a user's ability to interact with your page.

Additionally, TBT correlates well with the field metric [First Input Delay](https://web.dev/fid/)
(FID), which is a Core Web Vital.

## Performance score update  {: #score }

The [performance score in Lighthouse](https://web.dev/performance-scoring/) is calculated from a
weighted blend of multiple metrics to summarize a page's speed. The 6.0 performance score formula
follows.

<table>
<thead>
<tr>
<th><strong>Phase</strong></th>
<th><strong>Metric Name</strong></th>
<th><strong>Metric Weight</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>Early (15%)</td>
<td>First Contentful Paint (FCP)</td>
<td>15%</td>
</tr>
<tr>
<td>Mid (40%)</td>
<td>Speed Index (SI)</td>
<td>15%</td>
</tr>
<tr>
<td></td>
<td>Largest Contentful Paint (LCP)</td>
<td>25%</td>
</tr>
<tr>
<td>Late (15%)</td>
<td>Time To Interactive (TTI)</td>
<td>15%</td>
</tr>
<tr>
<td>Main Thread (25%)</td>
<td>Total Blocking Time (TBT)</td>
<td>25%</td>
</tr>
<tr>
<td>Predictability (5%)</td>
<td>Cumulative Layout Shift (CLS)</td>
<td>5%</td>
</tr>
</tbody>
</table>

> While three new metrics have been added, three old ones have been removed: First Meaningful Paint,

Reviewer: It does feel like we need a graphic here. Shall we do the 5->6 graphic we did in keynote? I can work with @egsweeny to do this.

Author: Is this just the TODO already marked below?

Reviewer: ![lh-weighting-change-v5-v6](https://user-images.githubusercontent.com/39191/82247717-b8088200-98fb-11ea-8c1b-53ebc3d54d60.png)

This could be better, but we'll use this for now.

   With this scoring change, we hope developers are more effectively encouraged to optimize for
   user interactivity.
-  **FCP's weight has been reduced from 23% to 15%.** Measuring only when the first pixel is
   painted (FCP) didn't give us a complete picture. Combining it with measuring when users are able
   to see what they most likely care about (LCP) better reflects the loading experience.
-  **Max Potential FID** **has been deprecated**. It is no longer shown in the report, but is
   still available in the JSON. It is now recommended to look at TBT to quantify your interactivity
   instead of mpFID.
-  **First Meaningful Paint has been deprecated.** This metric was too variant and had no viable
   path to standardization as the implementation is specific to Chrome rendering internals. While
   some teams do find the FMP timing to be worthwhile on their site, the metric will not receive
   additional improvements.
-  **First CPU Idle has been deprecated** because it's not distinct enough from TTI. TBT and TTI
   are the go-to metrics for interactivity now.
-  CLS's weight is relatively low, though we expect to increase it in a future major version.

> **TODO(cjamcl)**: Make two images for v5/v6 scoring weights.

Reviewer: TODO

relatively few sites but can have sizable impacts in certain situations. Overall, ~8% of sites
experienced a score improvement due to metric implementation changes and ~4% of sites saw a score
decrease due to metric implementation changes. ~88% of sites were unaffected by these fixes. 

Individual score curve changes also impacted the overall score shifts although very slightly. We
periodically ensure that the score curve aligns with the observed metrics in the [HTTPArchive
dataset](http://httparchive.org/). Excluding sites affected by major implementation changes, minor
adjustments to the score curve for individual metrics improved the scores of ~3% of sites and
decreased the scores of ~4% of sites. ~93% of sites were unaffected by this change.

### Calculator {: #calculator }

We've published a [scoring calculator](https://googlechrome.github.io/lighthouse/scorecalc/) to help
you explore performance scoring. The calculator also gives you a comparison between version v5 and
v6 scores. When you run an audit with Lighthouse 6.0, the report comes with a link to the calculator
with your results populated.

> ![image](lighthouse-scorecalc.png)

Reviewer: this definitely needs a 1px border. it actually looks interactive with the screenshot right now.

Reviewer 2: see note above about `w-screenshot`

> Huge thanks to [Ana Tudor](https://twitter.com/anatudor) for the gauge upgrade!

Reviewer: can we do this with a figure/figcaption?

out all the work done so far and starts over, leading to poor experiences for the user. This new
audit verifies the page has a valid character encoding and it's defined early and up front.

## Lighthouse CI {: #ci }

At [CDS last November](https://web.dev/lighthouse-evolution-cds-2019/#lighthouse-ci-alpha-release)
we announced [Lighthouse CI](https://github.com/GoogleChrome/lighthouse-ci), the open source Node
CLI and server that tracks Lighthouse results on every commit in your continuous integration
pipeline, and we've come a long way since the alpha release. Lighthouse CI now has native support
for numerous CI providers including Travis, Circle, GitLab, and Github Actions. Ready-to-deploy
[docker images](https://github.com/GoogleChrome/lighthouse-ci/tree/master/docs/recipes) make setup a
breeze, and a comprehensive dashboard redesign now reveals trends across every category and metric
in Lighthouse for detailed analysis.

Start using Lighthouse CI on your project today by following our
[getting started guide](https://github.com/GoogleChrome/lighthouse-ci/blob/master/docs/getting-started.md).

> ![image](lighthouse-ci-1.png)

Reviewer: i think these need a 1px border of some sort. dunno if moving them to be `<figure>`s would help but they're floating a bit too much

Reviewer 2: You can give items a 1px border by switching to an `<img class="w-screenshot">`. You can also wrap that `img` in a `figure` if you want it to have a caption.

Author: is there a click -> lightbox feature?

Reviewer 2: > is there a click -> lightbox feature?

unfortunately not

-  Screen size is changed from 412x660 px to 360x640 px.
-  The useragent changes slightly, just the device portion (previously "`Nexus 5 Build/MRA58N`"
   will now be `"Moto G (4)"`.

As of Chrome 81, Moto G4 is available in the Chrome DevTools device emulation list.

![image](lighthouse-motog4.png)

## Browser extension {: #extension }

The
[Chrome extension for Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk)
has been a convenient way to run Lighthouse locally. Unfortunately, it was complicated to support.
We felt that because the Chrome DevTools Lighthouse panel is a better experience (the report
integrates with other panels), we could reduce our engineering overhead by simplifying the Chrome
extension.

Instead of running Lighthouse locally, the extension now uses the [PageSpeed Insights
API](https://developers.google.com/speed/docs/insights/v5/get-started). We recognize that this will
not be a sufficient replacement for some of our users. These are the key differences:

-  Pagespeed Insights is unable to audit non-public websites, since it is run via a remote
   server and not your local Chrome instance. 

>    -  If you need to audit a non-public website, use the DevTools Lighthouse panel, or the

Reviewer: the spacing in the rendered thing kinda sucks. i think we should reformat this.

![image](https://user-images.githubusercontent.com/39191/82106746-bd858280-96d7-11ea-95b6-98bcb58b74db.png)

Reviewer 2: hrmmmmmmmmmm yeah that's a lot of spacing... If y'all would like to keep the nested list I can look into toning down that whitespace.

Reviewer 3: I've suggested a formatting change to fix this, though it would be great to tweak the list CSS a bit in the future :)

Lighthouse 5.0 introduced [performance budgets](https://web.dev/performance-budgets-101/) which
supported adding thresholds for
[how much of each resource type](https://github.com/GoogleChrome/lighthouse/blob/master/docs/performance-budgets.md#resource-budgets)
(such as scripts, images, or css) a page can serve.

Lighthouse 6.0 adds
[support for budgeting metrics](https://github.com/GoogleChrome/lighthouse/blob/master/docs/performance-budgets.md#timing-budgets),
so now you can set thresholds for specific metrics such as FCP. For now, budgets are only available
to the Node CLI and Lighthouse CI.

## Source location links {: #source-location }

Some of the issues that Lighthouse finds about a page can be traced back to a specific line of
source code and the report will state the exact file and line that's relevant. To make this easy to
explore in DevTools, clicking on the locations specified in the report will open the relevant files
in the **Sources** panel.

<figure>
  <video autoplay loop muted playsinline>
    <source src="./lighthouse-source-location.webm" type="video/webm">
    <source src="./lighthouse-source-location.mp4" type="video/mp4">
  </video>
  <figcaption>

>     DevTools reveals the exact line of code that causes the issue.

Reviewer: this video looks GREAT.

i feel like i saw figcaptions that were centered on web.dev. is that a convention?

Reviewer 2: yeah, use `w-figure` on the `<figure>` and `w-figcaption` on the `<figcaption>`

  </figcaption>
</figure>

## On the horizon {: #horizon }

Lighthouse has begun experimenting with collecting source maps to power new features, such as:

-  Detecting duplicate modules in JavaScript bundles.
-  Detecting excessive polyfills or transforms in code sent to modern browsers.
-  Augmenting the Unused JavaScript audit to provide module-level granularity.
-  Treemap visualizations highlighting the modules that require action.
-  Displaying the original source code for report items with a "source location".

> ![image](lighthouse-sourcemaps-unusedjs.png)

Reviewer: because the matte is white i think this needs a border.

also... maybe just delete a few rows in here with devtools so it can be shorter. feels a bit too tall.

Reviewer 2: may also need alt text for the image

The Unused JavaScript audit using source maps to show unused code in specific bundled modules

> These features will be enabled by default in a future version of Lighthouse. For now, you can view

Reviewer: ```suggestion
These features will be enabled by default in a future version of Lighthouse. In v6, you can view
```

ye?

Author: eh, I do plan for this to land in 6.1 or 6.2. `In v6.0 ... `?

```bash
npm install -g lighthouse
lighthouse https://www.example.com --view
```

This version of Lighthouse comes with a large number of changes that are all listed in the
changelog. (**TODO: link to changelog)**. We'll cover the highlights in this article.

- [New metrics](#new-metrics)
- [Performance score update](#score)
- [New audits](#new-audits)
- [Lighthouse CI](#ci)
- [Renamed Chrome DevTools panel](#devtools)
- [Mobile emulation](#emulation)
- [Browser extension](#extension)
- [Budgets](#budgets)
- [Source location links](#source-location)
- [On the horizon](#horizon)
- [Thank you!](#thanks)

## New metrics {: #new-metrics }

![image](lighthouse-new-metrics.png)

Lighthouse 6.0 introduces three new metrics to the report. Two of these new metrics–Largest
Contentful Paint (LCP) and Cumulative Layout Shift (CLS)–are lab implementations of [Core Web
Vitals](https://web.dev/vitals/).

### Largest Contentful Paint (LCP) {: #lcp }

[Largest Contentful Paint (LCP)](https://www.web.dev/lcp/) is a measurement of perceived loading
experience. It marks the point during page load when the primary–or "largest"–content has loaded and

> is visible to the user. LCP is an important complement to First Contentful Paint (FCP) which only

Author: ```suggestion
is visible to the user. LCP is an important complement to First Contentful Paint (FCP), which only
```

Author: the comma disambiguates what "which only" refers to. a rewrite would be better than no comma

---
title: "What's New in Lighthouse 6.0"
authors:

>   - cjamcl

Author: trying to figure out the @2x stuff

out all the work done so far and starts over, leading to poor experiences for the user. This new
audit verifies the page has a valid character encoding and it's defined early and up front.

## Lighthouse CI {: #ci }

At [CDS last November](https://web.dev/lighthouse-evolution-cds-2019/#lighthouse-ci-alpha-release)
we announced [Lighthouse CI](https://github.com/GoogleChrome/lighthouse-ci), the open source Node
CLI and server that tracks Lighthouse results on every commit in your continuous integration
pipeline, and we've come a long way since the alpha release. Lighthouse CI now has native support
for numerous CI providers including Travis, Circle, GitLab, and Github Actions. Ready-to-deploy
[docker images](https://github.com/GoogleChrome/lighthouse-ci/tree/master/docs/recipes) make setup a
breeze, and a comprehensive dashboard redesign now reveals trends across every category and metric
in Lighthouse for detailed analysis.

Start using Lighthouse CI on your project today by following our
[getting started guide](https://github.com/GoogleChrome/lighthouse-ci/blob/master/docs/getting-started.md).

> <figure class="w-figure">

Reviewer: super nitpicky but i think these images take up lot of screen real estate

i'd personally do something like this

![image](https://user-images.githubusercontent.com/39191/82240265-f64b7480-98ee-11ea-91ef-21e33631c71c.png)

and i'd link each image to the fullsize of itself.

  <img class="w-screenshot" src="lighthouse-ci-1.png" alt="Lighthouse CI." width="600">
</figure>
<figure class="w-figure">
  <img class="w-screenshot" src="lighthouse-ci-2.png" alt="Lighthouse CI." width="600">
</figure>
<figure class="w-figure">
  <img class="w-screenshot" src="lighthouse-ci-3.png" alt="Lighthouse CI." width="600">
</figure>

## Renamed Chrome DevTools panel {: #devtools }

> We've renamed the Audits panel to the Lighthouse panel. Enough said!

Reviewer: ```suggestion
We've renamed the `Audits` panel to the `Lighthouse` panel. Enough said!
```

suggestion from @egsweeny

   will now be `"Moto G (4)"`.

As of Chrome 81, Moto G4 is available in the Chrome DevTools device emulation list.

![image](lighthouse-motog4.png)

## Browser extension {: #extension }

The
[Chrome extension for Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk)
has been a convenient way to run Lighthouse locally. Unfortunately, it was complicated to support.
We felt that because the Chrome DevTools Lighthouse panel is a better experience (the report
integrates with other panels), we could reduce our engineering overhead by simplifying the Chrome
extension.

Instead of running Lighthouse locally, the extension now uses the [PageSpeed Insights
API](https://developers.google.com/speed/docs/insights/v5/get-started). We recognize that this will
not be a sufficient replacement for some of our users. These are the key differences:

-  Pagespeed Insights is unable to audit non-public websites, since it is run via a remote
   server and not your local Chrome instance. 
   -  If you need to audit a non-public website, use the DevTools Lighthouse panel, or the
      Node CLI.

> -  Pagespeed Insights is not guaranteed to use the latest Lighthouse version.

Reviewer: carrying over the thread from the doc. this is still so weird to me.   looking at someone else's project i would not consider "latest version" to ever mean `master` branch.  but regardless, I imagine everyone would be fine if we say  "release" instead of "version" to deambiguify. 

how about

```md
- PageSpeed Insights may not be using the latest Lighthouse release.
   - In general, PageSpeed Insights will be updated to the latest Lighthouse release within 1-2 weeks. Use the Node CLI if you want to run a specific Lighthouse version.
```

   -  If you want to use the latest Lighthouse version, use the Node CLI. The browser
      extension will get the update ~1-2 weeks after release.

> -  Pagespeed Insights is a Google API, using it constitutes accepting the Google API Terms of

Reviewer: s/Pagespeed/PageSpeed/g

   Service.
   -  If you do not wish to/cannot accept the terms of service, use the DevTools Lighthouse

>       panel, or the Node CLI.

Reviewer: yeah i think this is an improvement. (i nested the lists to begin with)

lgtm on this tho i have another comment which revises the 2nd bullet's text.

  - lighthouse
---

Today we're releasing Lighthouse 6.0!

[Lighthouse](https://github.com/GoogleChrome/lighthouse/) is an automated website auditing tool that
helps developers with opportunities and diagnostics to improve the user experience of their sites.
It's available in Chrome DevTools, npm (as a Node module and a CLI), or as a browser extension (in
[Chrome](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk) and
[Firefox](https://addons.mozilla.org/en-US/firefox/addon/google-lighthouse/)). It powers many Google
services, including [web.dev/measure](https://web.dev/measure/) and [PageSpeed
Insights](https://developers.google.com/speed/pagespeed/insights/).

Lighthouse 6.0 is available immediately on npm and in [Chrome
Canary](https://www.google.com/chrome/canary/). Other Google services that leverage Lighthouse will
get the update by the end of the month. It will land in Chrome Stable in Chrome 84 (mid-July).

To try the Lighthouse Node CLI, use the following commands:  
```bash
npm install -g lighthouse
lighthouse https://www.example.com --view
```

This version of Lighthouse comes with a large number of changes that are all listed in the

> changelog. (**TODO: link to changelog)**. We'll cover the highlights in this article.

Reviewer: https://github.com/GoogleChrome/lighthouse/releases/tag/v6.0.0

`[listed in the 6.0 changelog]`  probably

- [Renamed Chrome DevTools panel](#devtools)
- [Mobile emulation](#emulation)
- [Browser extension](#extension)
- [Budgets](#budgets)
- [Source location links](#source-location)
- [On the horizon](#horizon)
- [Thank you!](#thanks)

## New metrics {: #new-metrics }

<figure class="w-figure">
  <img class="w-screenshot" src="lighthouse-new-metrics.png" alt="Lighthouse 6.0 metrics." width="600">
</figure>

Lighthouse 6.0 introduces three new metrics to the report. Two of these new metrics–Largest
Contentful Paint (LCP) and Cumulative Layout Shift (CLS)–are lab implementations of [Core Web
Vitals](https://web.dev/vitals/).

### Largest Contentful Paint (LCP) {: #lcp }

[Largest Contentful Paint (LCP)](https://www.web.dev/lcp/) is a measurement of perceived loading
experience. It marks the point during page load when the primary–or "largest"–content has loaded and
is visible to the user. LCP is an important complement to First Contentful Paint (FCP) which only
captures the very beginning of the loading experience. LCP provides a signal to developers about how
quickly a user is actually able to see the content of a page. An LCP score below 2.5 seconds is
considered 'Good.'

> For more information, [watch this deep-dive on LCP](https://youtu.be/diAc65p15ag) by Paul Irish:

Reviewer: ```suggestion
For more information, [watch this deep-dive on LCP](https://youtu.be/diAc65p15ag) by Paul Irish.
```

### Cumulative Layout Shift (CLS) {: #cls }

[Cumulative Layout Shift (CLS)](https://www.web.dev/cls/) is a measurement of visual stability. It
quantifies how much a page's content visually shifts around. A low CLS score is a signal to
developers that their users aren't experiencing undue content shifts; a CLS score below 0.10 is
considered 'Good.'

CLS in a lab environment is measured through the end of page load. Whereas in the field, you can
measure CLS up to the first user interaction or including all user input.

> For more information, [watch this deep-dive on CLS](https://youtu.be/zIJuY-JCjqw) by Annie Sullivan:

Reviewer: ```suggestion
For more information, [watch this deep-dive on CLS](https://youtu.be/zIJuY-JCjqw) by Annie Sullivan.
```

periodically ensure that the score curve aligns with the observed metrics in the [HTTPArchive
dataset](http://httparchive.org/). Excluding sites affected by major implementation changes, minor
adjustments to the score curve for individual metrics improved the scores of about 3% of sites and
decreased the scores of about 4% of sites. Approximately 93% of sites were unaffected by this change.

### Scoring calculator {: #calculator }

We've published a [scoring calculator](https://googlechrome.github.io/lighthouse/scorecalc/) to help
you explore performance scoring. The calculator also gives you a comparison between Lighthouse version 5 and
6 scores. When you run an audit with Lighthouse 6.0, the report comes with a link to the calculator
with your results populated.

<figure class="w-figure">
  <img class="w-screenshot" src="lighthouse-scorecalc.png" alt="Lighthouse Score Calculator." width="600">
  <figcaption class="w-figcaption">Huge thanks to <a href="https://twitter.com/anatudor">Ana Tudor</a> for the gauge upgrade!</figcaption>
</figure>

## New audits {: #new-audits }

### Unused JavaScript {: #unused-javascript }

We are leveraging [DevTools code

> coverage](https://developers.google.com/web/tools/chrome-devtools/coverage) in a new audit: [Unused
> JavaScript](https://web.dev/remove-unused-code/).

Reviewer 3: ```suggestion
coverage](https://developers.google.com/web/tools/chrome-devtools/coverage) in a new audit: [**Unused
JavaScript**](https://web.dev/remove-unused-code/).
```

This audit isn't _entirely_ new: it was [added in
mid-2017](https://github.com/GoogleChrome/lighthouse/issues/1852#issuecomment-306900595), but
because of the performance overhead it was disabled by default to keep the Lighthouse as fast as
possible. Collecting this coverage data is far more efficient now, so we feel comfortable enabling
it by default.

### Accessibility audits {: #a11y }

Lighthouse uses the wonderful [axe-core](https://github.com/dequelabs/axe-core) library to power the
accessibility category. In Lighthouse 6.0, we've added the following audits:

-  [aria-hidden-body](https://web.dev/aria-hidden-body/)
-  [aria-hidden-focus](https://web.dev/aria-hidden-focus/)
-  [aria-input-field-name](https://web.dev/aria-input-field-name/)
-  [aria-toggle-field-name](https://web.dev/aria-toggle-field-name/)
-  [form-field-multiple-labels](https://web.dev/form-field-multiple-labels/)
-  [heading-order](https://web.dev/heading-order/)
-  [duplicate-id-active](https://web.dev/duplicate-id-active/)
-  [duplicate-id-aria](https://web.dev/duplicate-id-aria/)

> 

Reviewer 3: I'm bringing this up late, but could we use the full names of the audits in sentence case instead of the url shorthands?

[docker images](https://github.com/GoogleChrome/lighthouse-ci/tree/master/docs/recipes) make setup a
breeze, and a comprehensive dashboard redesign now reveals trends across every category and metric
in Lighthouse for detailed analysis.

Start using Lighthouse CI on your project today by following our
[getting started guide](https://github.com/GoogleChrome/lighthouse-ci/blob/master/docs/getting-started.md).

<style>
@media (min-width: 865px) {
  .w-figure--inline-left {
    margin: 0 20px 40px 0;
  }
}
</style>

<figure class="w-figure w-figure--inline-left">
  <a href="lighthouse-ci-1.png" target="_blank">
    <img class="w-screenshot" src="lighthouse-ci-1.png" alt="Lighthouse CI." width="600">
  </a>
</figure>
<figure class="w-figure w-figure--inline-left">
  <a href="lighthouse-ci-2.png" target="_blank">
    <img class="w-screenshot" src="lighthouse-ci-2.png" alt="Lighthouse CI." width="600">
  </a>
</figure>
<figure class="w-figure">
  <a href="lighthouse-ci-3.png" target="_blank">
    <img class="w-screenshot" src="lighthouse-ci-3.png" alt="Lighthouse CI." width="600">
  </a>
</figure>

## Renamed Chrome DevTools panel {: #devtools }

> We've renamed the `Audits` panel to the `Lighthouse` panel. Enough said!

Reviewer 3: ```suggestion
We've renamed the **Audits** panel to the **Lighthouse** panel. Enough said!
```

Reviewer 3: [Bold formatting for UI elements](https://web.dev/handbook/grammar/#ui-elements-and-interaction) is web.dev convention, sorry I've missed it in the first pass.

Depending on your DevTools window size, the panel is probably behind the `»` button. You can drag
the tab to change the order.

To quickly reveal the panel with the [Command
menu](https://developers.google.com/web/tools/chrome-devtools/command-menu): 

> 1. {% Instruction 'devtools' %}

Reviewer: don't know why but the formatting here got odd

![image](https://user-images.githubusercontent.com/39191/82251822-c7d79480-9902-11ea-97a3-0c56f2deb114.png)

Reviewer 3: My bad, I know what it is 🤦🏻‍♀️ I've suggested a fix 👍

> 1. {% Instruction 'devtools' %}

Reviewer 3: ```suggestion
{% Instruction 'devtools' %}
```

Author: I went with passing `'none`'

> 1. {% Instruction 'devtools-command', 'ol' %}

Reviewer 3: ```suggestion
{% Instruction 'devtools-command', 'ol' %}
```

   [Lantern](https://github.com/GoogleChrome/lighthouse/blob/master/docs/lantern.md)).
-  Device screen emulation (the same found in Chrome DevTools).

Since the beginning, Lighthouse has used Nexus 5X as its reference device. In recent years, most
performance engineers have been using Moto G4 for testing purposes. Now Lighthouse is following suit
and has changed its reference device to Moto G4. In practice, this change is not very noticeable,
but here are all the changes detectable by a webpage:

-  Screen size is changed from 412x660 px to 360x640 px.
-  The user agent string is changed slightly, the device portion that was previously `Nexus 5 Build/MRA58N`
   will now be `Moto G (4)`.

As of Chrome 81, Moto G4 is also available in the Chrome DevTools device emulation list.

![Chrome DevTools device emulation list with Moto G4 included.](lighthouse-motog4.png)

## Browser extension {: #extension }

The
[Chrome extension for Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk)
has been a convenient way to run Lighthouse locally. Unfortunately, it was complicated to support.

> We felt that because the Chrome DevTools Lighthouse panel is a better experience (the report

Reviewer 3: ```suggestion
We felt that because the Chrome DevTools **Lighthouse** panel is a better experience (the report
```

integrates with other panels), we could reduce our engineering overhead by simplifying the Chrome
extension.

Instead of running Lighthouse locally, the extension now uses the [PageSpeed Insights
API](https://developers.google.com/speed/docs/insights/v5/get-started). We recognize that this will
not be a sufficient replacement for some of our users. These are the key differences:

-  PageSpeed Insights is unable to audit non-public websites, since it is run via a remote
   server and not your local Chrome instance. If you need to audit a non-public website, use

>    the DevTools Lighthouse panel, or the Node CLI.

Reviewer 3: ```suggestion
   the DevTools **Lighthouse** panel, or the Node CLI.
```

-  PageSpeed Insights is not guaranteed to use the latest Lighthouse release. If you want to use
   the latest release, use the Node CLI. The browser extension will get the update ~1-2 weeks after release.
-  PageSpeed Insights is a Google API, using it constitutes accepting the Google API Terms of

>    Service. If you do not wish to/cannot accept the terms of service, use the DevTools Lighthouse panel,

Reviewer 3: ```suggestion
   Service. If you do not wish to or cannot accept the terms of service, use the DevTools **Lighthouse** panel,
```

-  Detecting duplicate modules in JavaScript bundles.
-  Detecting excessive polyfills or transforms in code sent to modern browsers.
-  Augmenting the Unused JavaScript audit to provide module-level granularity.
-  Treemap visualizations highlighting the modules that require action.
-  Displaying the original source code for report items with a "source location".

<figure class="w-figure">
  <img class="w-screenshot" src="lighthouse-sourcemaps-unusedjs.png" alt="Unused JavaScript showing modules from source maps." width="600">
  <figcaption class="w-caption">
    The Unused JavaScript audit using source maps to show unused code in specific bundled modules.
  </figcaption>
</figure>

These features will be enabled by default in a future version of Lighthouse. For now, you can view
Lighthouse's experimental audits with the following CLI flag:  
  
```bash
lighthouse https://web.dev --view --preset experimental
```

## Thank you! {: #thanks }

We thank you for using Lighthouse and providing feedback. Your feedback helps us improve Lighthouse
and we hope Lighthouse 6.0 will make it easier for you to improve the performance of your
websites.

What can you do next?

> -  Open Chrome Canary and give the Lighthouse panel a go.

Reviewer 3: ```suggestion
-  Open Chrome Canary and give the **Lighthouse** panel a go.
```

not be a sufficient replacement for some of our users. These are the key differences:

-  PageSpeed Insights is unable to audit non-public websites, since it is run via a remote
   server and not your local Chrome instance. 
   -  If you need to audit a non-public website, use the DevTools Lighthouse panel, or the
      Node CLI.

-  PageSpeed Insights is not guaranteed to use the latest Lighthouse release.
   -  If you want to use the latest release, use the Node CLI. The browser
      extension will get the update ~1-2 weeks after release.

-  PageSpeed Insights is a Google API, using it constitutes accepting the Google API Terms of
   Service.
   -  If you do not wish to/cannot accept the terms of service, use the DevTools Lighthouse
      panel, or the Node CLI.

The good news is that simplifying the product story allowed us to focus on other engineering
problems. As a result, we released the [Lighthouse Firefox
extension](https://addons.mozilla.org/en-US/firefox/addon/google-lighthouse/)!

## Budgets {: #budgets }

Lighthouse 5.0 introduced [performance budgets](https://web.dev/performance-budgets-101/) which

> supported adding thresholds for

Author: ```suggestion
supports setting thresholds for
```

---
title: "What's New in Lighthouse 6.0"
subhead: New metrics, Performance score update, new audits, and more.
authors:
  - cjamcl
date: 2020-05-19
hero: lighthouse-logo.svg
alt: Lighthouse logo.
tags:
  - blog # blog is a required tag for the article to show up in the blog.

>   - fast

Reviewer 2: ```suggestion
  - performance
```

considered 'Good.'

CLS in a lab environment is measured through the end of page load. Whereas in the field, you can
measure CLS up to the first user interaction or including all user input.

For more information, [watch this deep-dive on CLS](https://youtu.be/zIJuY-JCjqw) by Annie Sullivan.

### Total Blocking Time (TBT) {: #tbt }

[Total Blocking Time (TBT)](https://www.web.dev/tbt/) quantifies load responsiveness, measuring the
total amount of time when the main thread was blocked long enough to prevent input responsiveness.
TBT measures the total amount of time between First Contentful Paint (FCP) and Time to Interactive
(TTI). It is a companion metric to TTI and it brings more nuance to quantifying main thread activity
that blocks a user's ability to interact with your page.

Additionally, TBT correlates well with the field metric [First Input Delay](https://web.dev/fid/)
(FID), which is a Core Web Vital.

## Performance score update  {: #score }

The [performance score in Lighthouse](https://web.dev/performance-scoring/) is calculated from a
weighted blend of multiple metrics to summarize a page's speed. The 6.0 performance score formula
follows.

> <style>

Reviewer 2: I was wondering why y'all needed this? Maybe because the table is missing the outer `w-table-wrapper` class?

.lh-table {
  min-width: unset;
}
.lh-table td {
  min-width: unset;
}
</style>

> <table class="lh-table">

Reviewer 2: wrap this table in `<div class="w-table-wrapper">` so it overflows properly on mobile. Might be able to remove the inline style.

<thead>
<tr>
<th><strong>Phase</strong></th>
<th><strong>Metric Name</strong></th>
<th><strong>Metric Weight</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>Early (15%)</td>
<td>First Contentful Paint (FCP)</td>
<td>15%</td>
</tr>
<tr>
<td>Mid (40%)</td>
<td>Speed Index (SI)</td>
<td>15%</td>
</tr>
<tr>
<td></td>
<td>Largest Contentful Paint (LCP)</td>
<td>25%</td>
</tr>
<tr>
<td>Late (15%)</td>
<td>Time To Interactive (TTI)</td>
<td>15%</td>
</tr>
<tr>
<td>Main Thread (25%)</td>
<td>Total Blocking Time (TBT)</td>
<td>25%</td>
</tr>
<tr>
<td>Predictability (5%)</td>
<td>Cumulative Layout Shift (CLS)</td>
<td>5%</td>
</tr>
</tbody>
</table>

While three new metrics have been added, three old ones have been removed: First Meaningful Paint,
First CPU Idle, and Max Potential FID. The weights of remaining metrics have been modified to
emphasize main thread interactivity and layout predictability.

For comparison, here is version 5 scoring:

> <table class="lh-table">

Reviewer 2: wrap this table in `<div class="w-table-wrapper">`

<thead>
<tr>
<th><strong>Phase</strong></th>
<th><strong>Metric Name</strong></th>
<th><strong>Weight</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>Early (23%)</td>
<td>First Contentful Paint (FCP)</td>
<td>23%</td>
</tr>
<tr>
<td>Mid (34%)</td>
<td>Speed Index (SI)</td>
<td>27%</td>
</tr>
<tr>
<td></td>
<td>First Meaningful Paint (FMP)</td>
<td>7%</td>
</tr>
<tr>
<td>Finished  (46%)</td>
<td>Time to Interactive (TTI)</td>
<td>33%</td>
</tr>
<tr>
<td></td>
<td>First CPU Idle (FCI)</td>
<td>13%</td>
</tr>
<tr>
<td>Main Thread</td>
<td>Max Potential FID</td>
<td>0%</td>
</tr>
</tbody>
</table>

> ![Lighthouse scoring changes between versions 5 and 6.](lighthouse-weight-changes.png)

Reviewer 2: ```suggestion
<img class="w-screenshot" src="./lighthouse-weight-changes.png" alt="Lighthouse scoring changes between versions 5 and 6." />
```

out all the work done so far and starts over, leading to poor experiences for the user. This new
audit verifies the page has a valid character encoding and it's defined early and up front.

## Lighthouse CI {: #ci }

At [CDS last November](https://web.dev/lighthouse-evolution-cds-2019/#lighthouse-ci-alpha-release)
we announced [Lighthouse CI](https://github.com/GoogleChrome/lighthouse-ci), the open source Node
CLI and server that tracks Lighthouse results on every commit in your continuous integration
pipeline, and we've come a long way since the alpha release. Lighthouse CI now has native support
for numerous CI providers including Travis, Circle, GitLab, and Github Actions. Ready-to-deploy
[docker images](https://github.com/GoogleChrome/lighthouse-ci/tree/master/docs/recipes) make setup a
breeze, and a comprehensive dashboard redesign now reveals trends across every category and metric
in Lighthouse for detailed analysis.

Start using Lighthouse CI on your project today by following our
[getting started guide](https://github.com/GoogleChrome/lighthouse-ci/blob/master/docs/getting-started.md).

> <style>

Reviewer 2: Might consider removing this and letting the images be 100% (or near 100%) width. On desktop it's hard to make out the contents of the top two screenshots.

explore in DevTools, clicking on the locations specified in the report will open the relevant files
in the **Sources** panel.

<figure class="w-figure">
  <video autoplay loop muted playsinline class="w-screenshot">
    <source src="https://storage.googleapis.com/web-dev-assets/lighthouse-whats-new-6.0/lighthouse-source-location.webm" type="video/webm">
    <source src="https://storage.googleapis.com/web-dev-assets/lighthouse-whats-new-6.0/lighthouse-source-location.mp4" type="video/mp4">
  </video>
  <figcaption class="w-caption">
    DevTools reveals the exact line of code that causes the issue.
  </figcaption>
</figure>

## On the horizon {: #horizon }

Lighthouse has begun experimenting with collecting source maps to power new features, such as:

-  Detecting duplicate modules in JavaScript bundles.
-  Detecting excessive polyfills or transforms in code sent to modern browsers.
-  Augmenting the Unused JavaScript audit to provide module-level granularity.
-  Treemap visualizations highlighting the modules that require action.
-  Displaying the original source code for report items with a "source location".

<figure class="w-figure">

>   <img class="w-screenshot" src="lighthouse-sourcemaps-unusedjs.png" alt="Unused JavaScript showing modules from source maps." width="600">

Reviewer 2: kinda hard to read the text on this page 🤓
might consider letting it be 100% width
